latte/latte.TESTGEN.ps1

```diff
@@ -77,99 +73,117 @@ function test_send {
-        [int] $tmp = 50000
-        [int] $sec = 30
-        foreach ($snd in $snds) {
-            for ($i=0; $i -lt $g_iters; $i++) {
-                [int] $portstart = $tmp + ($i * $g_iters)
-
-                # Optimized
-                test_send -Iter "-t $sec" -Port $portstart -Type "-$soc" -Snd $snd -Options "-group 0 -rio -riopoll 100000000000" -OutDir $dirOptimized -Fname "$soc.t$sec.$snd.OPT.iter$i" -NoDumpParam $true
-                test_recv
-                
-                # Default
-                if ($g_detail) {
-                    test_send -Iter "-t $sec" -Port $portstart -Type "-$soc" -Snd $snd -OutDir $dirDefault -Fname "$soc.t$sec.$snd.iter$i" -NoDumpParam $true
-                    test_recv
-                }
-            }
+        if ($g_Config.Time -gt 0) {
+            banner -Msg "Time Tests: [$Protocol] operations per bounded time"
+            test_protocol -Iter "-t $($g_Config.Time)" -Protocol $Protocol -OutDirOpt $dirOptimized -OutDirDefault $dirDefault -Fname "$Protocol.t$($g_Config.Time)" -NoDumpParam $true
         }
     }
 } # test_latte_generate()
 
+function validate_config {
+    $isValid = $true
+    $int_vars = @('Iterations', 'StartPort', 'Time', 'PingIterations')
+    foreach ($var in $int_vars) {
+        if (($null -eq $g_Config.($var)) -or ($g_Config.($var) -lt 0)) {
+            Write-Host "$var is required and must be greater than or equal to 0"
+            $isValid = $false
+        }
+    }
+    $valid_protocols = @('tcp', 'udp', 'raw')
+    foreach ($proto in $g_Config.Protocol) {
+        if (-Not $valid_protocols.Contains($proto)) {
+            Write-Host "$proto is not a valid protocol"
+            $isValid = $false
+        }
+    }
+    $valid_send = @('b', 'nb', 'ove', 'ovc', 'ovp', 'sel')
+    foreach ($snd in $g_Config.SendMethod) {
+        if (-Not $valid_send.Contains($snd)) {
+            Write-Host "$snd is not a valid send method"
+            $isValid = $false
+        }
+    }
+    return $isValid
+} # validate_config()
+
 #===============================================
 # External Functions - Main Program
 #===============================================
 function test_main {
     Param(
-        [parameter(Mandatory=$false)] [switch] $Detail = $false,
-        [parameter(Mandatory=$false)] [Int]    $Iterations = 1,
-        [parameter(Mandatory=$false)] [ValidateSet('Sampling','Testing')] [string] $Config = "Sampling",
+        [parameter(Mandatory=$false)] [string] $Config = "Default",
         [parameter(Mandatory=$true)]  [string] $DestIp,
         [parameter(Mandatory=$true)]  [string] $SrcIp,
         [parameter(Mandatory=$true)]  [ValidateScript({Test-Path $_ -PathType Container})] [String] $OutDir = "" 
     )
     input_display
-    
-    [int]    $g_iters      = $Iterations
-    [bool]   $g_detail     = $Detail
+    # get config variables
+    $allConfig = Get-Content ./latte/latte.Config.json | ConvertFrom-Json
```

Comment:
seeking @omarcardona 's input here about relative paths per a prior offline conversation. Can we assume that commands will be run from the master directory?